scheduler/docs/groups.md

```diff
@@ -1,56 +1,106 @@
+group.  See the api docs for more details.
 
 ## Straggler handling
 
-When running a lot of jobs, occasionally it happens that one or two of them run much slower than the rest.
-This can be because they are on a box that is in a bad state, experiencing network problems or the process
-is in a bad state. To avoid having to manually monitor the jobs to find and address these problem cases,
-Cook can be configured to kill and restart jobs in a group that have been running much longer then the
-other jobs in the group. 
+When running a lot of jobs, occasionally it happens that one or two of them run much slower than the
+rest.  This can be because they are on a box that is in a bad state, experiencing network problems
+or the process is in a bad state. To avoid having to manually monitor the jobs to find and address
+these problem cases, Cook can be configured to kill and restart jobs in a group that have been
+running much longer then the other jobs in the group. 
 
 To configure this, set the straggler handling type to `quantile-deviation`. This will have Cook look
-for jobs that have been running multiple times longer than a particular quantile of the group. 
-The parameters, `quantile` and `multiplier` decide when a job is considered 
-a straggler. The `quantile` decides the target runtime of the group and the `multipier` 
-decides how many times longer than the run time of the quantile job before a job is marked as a 
-straggler.
-
-
-
-  
-
+for jobs that have been running multiple times longer than a particular quantile of the group. The
+parameters, `quantile` and `multiplier` decide when a job is considered a straggler. The
+`quantile` decides the target runtime of the group and the `multipier` decides how many times
+longer than the run time of the quantile job before a job is marked as a straggler.
+
+## Host placement constraints
+
+Sometimes, it is desirable to set some rules on the host distribution to be used for the jobs in a
+group.  For example, jobs in a group might run faster if they are all put on different hosts.
+Another set of jobs may benefit from running on hosts that share the same attribute, such as
+availability zone. Job groups can be configured with host placement constraints to fulfill this
+requirement. Currently, there are three types of host placement constraints:
+
+### Unique host constraint
+Setting this constraint forces the scheduler to place each job in the group on a different machine
+(based on hostname).
+Example:
+```json
+{"groups": [{"uuid": "410198dd-c171-470e-aeb8-6ffa4e6a2ada",
+            "host-placement": {"type": "unique"}],
+ "jobs": [{"uuid": "831836ee-d5c3-47f7-a052-f4c3ce502495",
+           "group": "410198dd-c171-470e-aeb8-6ffa4e6a2ada". ...},
+           ...]}}
+```
+
+
+### Balanced host attribute constraint
+This constraint takes two parameters: an attribute name and a minimum spread. When using it, the
```

Comment:
There are no restrictions coming from Cook, so the attribute can be any attribute recognized by Mesos (which I assume supports setting custom attributes on your machine). @wyegelwel, is this correct?